app/models/spree/stock_location_decorator.rb

```diff
@@ -1,12 +1,12 @@
 Spree::StockLocation.class_eval do
 
-  belongs_to :supplier
+  belongs_to :supplier, class_name: 'Spree::Supplier'
 
   scope :by_supplier, -> (supplier_id) { where(supplier_id: supplier_id) }
 
   # Wrapper for creating a new stock item respecting the backorderable config and supplier
-  durably_decorate :propagate_variant, mode: 'soft', sha: 'f35b0d8a811311d4886d53024a9aa34e3aa5f8cb' do |variant|
-    if self.supplier_id.nil? or self.supplier_id == variant.supplier_id
+  durably_decorate :propagate_variant, mode: 'soft', sha: '68fd322da48facbdbf567a2391c2495d04c3c8bc' do |variant|
+    if self.supplier_id.blank? || variant.suppliers.pluck(:id).include?(self.supplier_id)
```

Comment:
you use this logic a few times, it might make sense to move it to an instance method on `Variant`
``` Ruby
def has_supplier?(supplier_id)
  suppliers.pluck(:id).include?(supplier_id)
end
```